### PR TITLE
Split microphysics tests into 0, 1, 2 -moment and nonequilibrium

### PR DESCRIPTION
**Pull request by trontrytel**

### CODE REVIEW

File: test/microphysics0M_tests.jl [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+import CloudMicrophysics.Microphysics0M as CM0
+
+include(joinpath(pkgdir(CM), "test", "create_parameters.jl"))
+
+@info "0-moment microphysics tests"
+
+function test_microphysics0M(FT)
+
+    toml_dict = CP.create_toml_dict(FT; dict_type = "alias")
+    prs = cloud_microphysics_parameters(toml_dict)
```
Comment (sriharshakandala) [inline test/microphysics0M_tests.jl:17]: Do we still need this?

File: test/microphysics0M_tests.jl [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+        TT.@test CM0.remove_precipitation(params_0m, q, q_vap_sat) ≈ FT(0)
+
+        # rain based on qc threshold
+        for lf in frac
+            q_liq = qc * lf
+            q_ice = (1 - lf) * qc
+
+            q = TD.PhasePartition(q_tot, q_liq, q_ice)
+
+            TT.@test CM0.remove_precipitation(prs, q) ≈
```
Comment (sriharshakandala) [inline test/microphysics0M_tests.jl:40]: Can we use 'params_0m` here?

File: test/microphysics0M_tests.jl (hunk 1 above)
Comment (trontrytel) [reply test/microphysics0M_tests.jl:17]: Will be fixed in the next PR

File: test/microphysics0M_tests.jl (hunk 2 above)
Comment (trontrytel) [reply test/microphysics0M_tests.jl:40]: Will be fixed in the next PR

**Pull request merged**